### PR TITLE
Introduced client-based API

### LINKED ISSUE #12: Way of supporting client-based API
As you might know, there some API calls which must be performed against particular client (files list, file's content etc).

What do you think about having [such feature](https://github.com/smoope/nomad-api/blob/feature/members/src/main/java/io/github/zanella/nomad/NomadClient.java#L32) in your lib (improvements like caching must be added)? 
The call will be `client.node(someNodeId).v1.client.getFileList(path)`.

In general it could be done at client side as well, but then it will have to deal with managing pool of such clients.. While doing it in SDK makes it much easier since the pool of members already part of client itself.

### PR DESCRIPTION
**Pull request by vicmosin**
Introduced file read content support.
Resolves #12

### CODE REVIEW

File: src/main/java/io/github/zanella/nomad/NomadClient.java [hunk 1]
```diff
@@ -24,4 +28,17 @@ public NomadClient(final String agentHost, final int agentPort) {
 
         v1 = new V1Client(agentHostWithScheme, agentPort);
     }
+
+    public NomadClient node(String nodeId) {
+        NodeInfo node = v1.node.getNode(nodeId);
+        if (node != null) {
+            Pattern pattern = Pattern.compile("\\s*(.*?):(\\d+)\\s*");
```
Comment (zanella) [inline src/main/java/io/github/zanella/nomad/NomadClient.java]: Can you move it up and make it `private static final` ?

File: src/main/java/io/github/zanella/nomad/NomadClient.java [hunk 2]
```diff
@@ -24,4 +28,17 @@ public NomadClient(final String agentHost, final int agentPort) {
 
         v1 = new V1Client(agentHostWithScheme, agentPort);
     }
+
+    public NomadClient node(String nodeId) {
+        NodeInfo node = v1.node.getNode(nodeId);
```
Comment (zanella) [inline src/main/java/io/github/zanella/nomad/NomadClient.java]: Can it be `final` ?

File: src/main/java/io/github/zanella/nomad/NomadClient.java [hunk 3]
```diff
@@ -24,4 +28,17 @@ public NomadClient(final String agentHost, final int agentPort) {
...
+        if (node != null) {
+            Pattern pattern = Pattern.compile("\\s*(.*?):(\\d+)\\s*");
+            Matcher matcher = pattern.matcher(node.getHttpAddr());
```
Comment (zanella) [inline src/main/java/io/github/zanella/nomad/NomadClient.java]: `final` too

File: src/main/java/io/github/zanella/nomad/v1/V1Client.java [hunk 4]
```diff
@@ -92,4 +96,21 @@ protected ObjectMapper customObjectMapper() {
                 .configure(SerializationFeature.WRAP_ROOT_VALUE, true)
                 .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
     }
+
+    static class JacksonDecoderExtended extends JacksonDecoder {
```
Comment (donovanmuller) [inline src/main/java/io/github/zanella/nomad/v1/V1Client.java]: Any particular reason this class is `static`?

Comment (vicmosin) [inline src/main/java/io/github/zanella/nomad/v1/V1Client.java]: Not really..

File: src/main/java/io/github/zanella/nomad/NomadClient.java (hunk 3 above)
Comment (vicmosin) [inline src/main/java/io/github/zanella/nomad/NomadClient.java]: Done

**Pull request merged**